sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -112,15 +116,31 @@ public synchronized void start(final @NotNull TracesSampler tracesSampler) {
       logger.log(SentryLevel.DEBUG, "Profiler was not started due to sampling decision.");
       return;
     }
-    if (isRunning()) {
-      logger.log(SentryLevel.DEBUG, "Profiler is already running.");
-      return;
+    switch (profileLifecycle) {
+      case TRACE:
+        // rootSpanCounter should never be negative, unless the user changed profile lifecycle while
+        // the profiler is running or close() is called. This is just a safety check.
+        if (rootSpanCounter < 0) {
```

Comment:
Why do you need keep track of counting here? Wouldn't it be easier to simply check if there's a running transaction or not? E.g. via `Sentry.getCurrentScopes().getTransaction()`

Reply:
the counter increases for any transaction, while `Sentry.getCurrentScopes().getTransaction()` returns only the transaction bound to the scope, right?

Reply:
Ah ok, my assumption was that this is being used only when a transaction is bound to scope. All good then :+1: